### PR TITLE
Moved the test files in a similar structure as the run-time code

### PR DESCRIPTION
**Pull request by dennisdoomen**
Create some sanity in that huge list of `.specs` files..

### CODE REVIEW

File: Tests/Shared.Specs/Shared.Specs.projitems [hunk 1]
```diff
@@ -9,81 +9,79 @@
...
-    <Compile Include="$(MSBuildThisFileDirectory)StringAssertionSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)TaskAssertionSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)TaskOfTAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Numeric\ComparableSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Numeric\NullableNumericAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Numeric\NumericAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\BooleanAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\DateTimeAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\DateTimeOffsetAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\DateTimeOffsetValueFormatterSpecs.cs" />
```
Comment (jnyrup) [inline Tests/Shared.Specs/Shared.Specs.projitems]: Move `DateTimeOffsetValueFormatterSpecs` and `TimeSpanFormatterSpecs` be moved to `Formatting/`?

File: Tests/Shared.Specs/Shared.Specs.projitems [hunk 2]
```diff
@@ -9,81 +9,79 @@
...
-    <Compile Include="$(MSBuildThisFileDirectory)TypeExtensionsSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)TypeSelectorSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XAttributeAssertionSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XAttributeFormatterSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XDocumentAssertionSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XDocumentFormatterSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XElementAssertionSpecs.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)XElementFormatterSpecs.cs" />
-  </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' != 'netcoreapp1.1' ">
```
Comment (jnyrup) [inline Tests/Shared.Specs/Shared.Specs.projitems]: The failing build is probably due to these files being conditionally included before.

File: Tests/Shared.Specs/Shared.Specs.projitems [hunk 3]
```diff
@@ -9,81 +9,79 @@
...
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\NullableBooleanAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\NullableEnumAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\NullableGuidAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\NullableSimpleTimeSpanAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\ObjectAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\ReferenceTypeAssertionsSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\SimpleTimeSpanAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\StringAssertionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\TimeSpanConversionExtensionSpecs.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Primitives\TimeSpanExtensions.cs" />
```
Comment (jnyrup) [inline Tests/Shared.Specs/Shared.Specs.projitems]: Move `TimeSpanConversionExtensionSpecs` and `TimeSpanExtensions` (and rename to `TimeSpanExtensionSpecs`) to `extensions/`?

Comment (dennisdoomen) [reply Tests/Shared.Specs/Shared.Specs.projitems]: `TimeSpanExtensions` is actually just code supporting the tests. Was thinking of having a `Common` folder or something to keep the root clean.

**Pull request merged**